### PR TITLE
OSD-18529 - HLO build prerequisites

### PR DESCRIPTION
**Pull request by Tafhim**
Resolves [OSD-18529](https://issues.redhat.com//browse/OSD-18529)
This PR adds prerequisites for onboarding HLO build pipeline to `app-interface`

### CODE REVIEW

File: build/build_package.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
```
Comment (samanthajayasinghe) [inline build/build_package.sh:1]: Maybe move sh file `script` or `bin` folder and keep the dockerfiles in the root folder

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    spec:
+      clusterDeploymentSelector:
+        matchLabels:
+          api.openshift.com/managed: "true"
+      resourceApplyMode: Sync
+      resources:
+        - apiVersion: v1
+          kind: Namespace
+          metadata:
+            name: openshift-hypershift-logging-operator
```
[the remark this answers is not in the document]
Comment (hectorakemp) [reply hack/olm-registry/olm-artifacts-template.yaml:45]: I think we just keep it as-is - following the convention of other operators (https://github.com/openshift/custom-domains-operator/blob/master/hack/olm-registry/olm-artifacts-template.yaml, https://github.com/openshift/cloud-ingress-operator/blob/master/hack/olm-registry/olm-artifacts-template.yaml)

File: build/build_package.sh (hunk 1 above)
Comment (hectorakemp) [reply build/build_package.sh:1]: :+1: - build/bin/*?

**Review by hectorakemp — COMMENTED** [review]
/lgtm

**Pull request merged**